wave_vision/include/wave/vision/tracker/impl/tracker.hpp

```diff
@@ -0,0 +1,239 @@
+#include "wave/vision/tracker/tracker.hpp"
+
+namespace wave {
+
+// Private Functions
+template <typename TDetector, typename TDescriptor, typename TMatcher>
+void Tracker<TDetector, TDescriptor, TMatcher>::detectAndCompute(
+  const cv::Mat &image,
+  std::vector<cv::KeyPoint> &keypoints,
+  cv::Mat &descriptor) {
+    keypoints = this->detector.detectFeatures(image);
+    descriptor = this->descriptor.extractDescriptors(image, keypoints);
+}
+
+template <typename TDetector, typename TDescriptor, typename TMatcher>
+std::map<int, size_t>
+Tracker<TDetector, TDescriptor, TMatcher>::registerKeypoints(
+  const std::vector<cv::KeyPoint> &curr_kp,
+  const std::vector<cv::DMatch> &matches) {
+    // Maps current keypoint indices to IDs
+    std::map<int, size_t> curr_ids;
+
+    // TODO will need way to specify camera sensor ID.
+    int sensor_id = 0;
+
+    for (const auto &m : matches) {
+        // Check to see if ID has already been assigned to keypoint
+        if (this->prev_ids.find(m.queryIdx) != this->prev_ids.end()) {
+            // If so, assign that ID to current map.
+            curr_ids[m.trainIdx] = this->prev_ids.at(m.queryIdx);
+
+            // Extract value of keypoint.
+            Vec2 landmark = convertKeypoint(curr_kp.at(m.trainIdx));
+
+            // Create LandmarkMeasurement, and add to
+            // LandmarkMeasurementContainer
+            LandmarkMeasurement<int> measurement(
+              this->img_times.at(this->img_count),
+              sensor_id,
+              curr_ids.at(m.trainIdx),
+              landmark);
+            this->landmarks.insert(measurement);
+        } else {
```

Comment:
I think with the latest changes it's looking a bit cleaner. Let me know what you think @leokoppel